internal/download/get.go

```diff
@@ -45,8 +45,12 @@ func httpGet(url string) ([]byte, *failures.Failure) {
 }
 
 func httpGetWithProgress(url string, progress *progress.Progress) ([]byte, *failures.Failure) {
-	logging.Debug("Retrieving url: %s", url)
-	client := retryhttp.NewClient(0 /* 0 = no timeout */, 5)
+	return httpGetWithProgressRetry(url, progress, 0, 3)
+}
+
+func httpGetWithProgressRetry(url string, progress *progress.Progress, attempt int, retries int) ([]byte, *failures.Failure) {
+	logging.Debug("Retrieving url: %s, attempt: %d", url, attempt)
+	client := retryhttp.NewClient(0 /* 0 = no timeout */, retries)
```

Comment:
The quantity of http connection retries expanding exponentially here is not obvious to callers. It's probably fine for now being hidden away behind "httpGetWithProgress", but maybe a comment is warranted.